lib/src/main/java/com/github/niqdev/openwebnet/OpenWebNetObservable.java

```diff
@@ -88,9 +88,17 @@ private OpenWebNetObservable() {}
         return context -> {
             try {
                 ByteBuffer buffer = context.getEmptyBuffer();
-                Integer count = context.getClient().read(buffer);
-                String message = new String(buffer.array()).trim();
-                log(String.format("read: %d|%s", count, message));
+                SocketChannel channel = context.getClient();
+                Integer count = 1;
+                String message = "";
+                
+                while(count >0 && !message.contains(ACK)) {
+                	int length = message.length();
```

Comment:
is this used only for printing purposes, right? if yes, please remove it and change this line
```
log(String.format("read: %d|%s", count, message.substring(message.length())));
```